src/test/java/com/netflix/playback/features/impl/PlaybackDiagnosticsImplTest.java

```diff
@@ -0,0 +1,104 @@
+package com.netflix.playback.features.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netflix.playback.features.model.PeriodCompletedCallback;
+import com.netflix.playback.features.model.PlaybackRequest;
+
+/**
+ * Tests {@link PlaybackDiagnosticsImpl}.
+ */
+public class PlaybackDiagnosticsImplTest {
+
+    private static Logger logger = LoggerFactory.getLogger(TimerImplTest.class);
+
+    private int periodLength = 100;
```

Comment:
The tests rely on wall clock time, specifically 100 milliseconds, and a latch to verify the results.  This may not be the best approach if we need to scale out the test suite by hundreds or thousands of tests.  How might you refactor the code so the tests are more scalable?